src/components/text-editor/__internal__/label-wrapper/label-wrapper.component.js

```diff
@@ -0,0 +1,19 @@
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+/* eslint-disable jsx-a11y/no-static-element-interactions */
+import React from 'react';
+import PropTypes from 'prop-types';
+/**
+ * TextEditor component is componsed with divs and spans.
+ * We have to manually trigger focus on TextEditor by clicking on label component.
+ * This wrapper allows us to trigger focus on TextEditor
+ */
+const LabelWrapper = ({ onClick, children }) => {
+  return <span onClick={ onClick }> {children} </span>;
+};
+
+LabelWrapper.propTypes = {
+  onClick: PropTypes.func,
```

Comment:
I think we should just make these required props even though they are internal and will never not be set. In reality we probably don't need `d.ts` files for anything but `text-editor.d.ts` but I added them so it's fine to continue.

```suggestion
  onClick: PropTypes.func.isRequired,
  children: PropTypes.node.isRequired
```